lib/Doctrine/ODM/PHPCR/Query/Builder/ConverterPhpcr.php

```diff
@@ -145,13 +145,17 @@ protected function getPhpcrProperty($originalAlias, $odmField)
 
         $propertyName = $fieldMeta['property'];
 
+        if ($fieldMeta['type'] == 'uuid') {
+            $propertyName = 'jcr:uuid';
+        }
+
         if (empty($fieldMeta['translated'])
             || empty($this->translator[$originalAlias])
         ) {
             return array($originalAlias, $propertyName);
         }
 
-        $propertyPath = $this->translator[$originalAlias]->getTranslatedPropertyPath($originalAlias, $fieldMeta['property'], $this->locale);
+        $propertyPath = $this->translator[$originalAlias]->getTranslatedPropertyPath($originalAlias, $propertyName, $this->locale);
```

Comment:
this change makes sense regardless of the rest, its weird that we extract the value into a variable but then only use it once.